modules/service/src/main/scala/lucuma/odb/service/ProgramService.scala

```diff
@@ -205,12 +222,14 @@ object ProgramService {
         ).flatten
       )
 
-    def updatePrograms(SET: ProgramPropertiesInput.Edit, which: AppliedFragment): Option[AppliedFragment] =
+    def updatePrograms(SET: ProgramPropertiesInput.Edit): Option[AppliedFragment] =
       updates(SET).map { us =>
-        void"UPDATE t_program " |+|
-        void"SET " |+| us.intercalate(void", ") |+| void" " |+|
-        void"WHERE t_program.c_program_id IN (" |+| which |+| void") " |+|
-        void"RETURNING t_program.c_program_id"
+        void"""
+          UPDATE t_program
+          SET """ |+| us.intercalate(void", ") |+| void"""
+          FROM t_program_update WHERE t_program.c_program_id = t_program_update.c_program_id
+          RETURNING t_program.c_program_id
+        """
```

Comment:
This seems good.  I'll do it in similar situations.  The only part that gives me pause is that elsewhere you have to assume that `t_program_update` exists already but nothing guarantees it?  This seems better than passing the list around though.